crates/ika-core/src/dwallet_mpc/dwallet_mpc_service.rs

```diff
@@ -183,69 +218,166 @@ impl DWalletMPCService {
                     .await;
             }
 
-            self.dwallet_mpc_manager
-                .handle_dwallet_db_message(DWalletMPCDBMessage::PerformCryptographicComputations)
-                .await;
-        }
-    }
-
-    /// Receive all completed MPC sessions from the MPC Output Verifier over the
-    /// `consensus_round_completed_sessions` channel.
-    /// If the session exists, mark is as [`MPCSessionStatus::Finished`].
-    /// Otherwise, create a new session with that status, to avoid re-running the computation for it.
-    fn receive_completed_mpc_session_identifiers(&mut self, bootstrap: bool) {
-        let mut completed_sessions = HashSet::new();
-        loop {
-            match self.consensus_round_completed_sessions_receiver.try_recv() {
-                Err(mpsc::error::TryRecvError::Empty) => {
-                    // No more completed sessions to report at the moment.
-                    break;
+            for (round, outputs) in mpc_outputs {
+                let mut messages = vec![];
+                let mut completed_sessions = vec![];
+                for output in outputs {
+                    let output_result = self
+                        .dwallet_mpc_manager
+                        .handle_dwallet_db_output(&output)
+                        .await;
+                    let session_identifier = output.session_request.session_identifier;
+                    match output_result {
+                        Ok(output_result) => match output_result.result {
+                            OutputVerificationStatus::FirstQuorumReached(m) => {
+                                messages.extend(m);
+                                completed_sessions.push(session_identifier);
+                                let output_digest = keccak256_digest(&output.output);
+                                info!(
+                                    ?output_digest,
+                                    round,
+                                    ?session_identifier,
+                                    "MPC output is verified and reached quorum"
+                                );
+                            }
+                            OutputVerificationStatus::Malicious => {
+                                debug!(
```

Comment:
```suggestion
                                warn!(
```

Reply:
though I'm not sure what malicious output means here

Reply:
Not sure what it means. Let's leave it for now and improve this code later